logging.add.async build fails on mtpm with xcc-clang

**Describe the bug**
tests/subsys/logging/log_core_additional/logging.add.async fails to build on intel_adsp_ace15_mtpm with XCC-clang toolchain.

**To Reproduce**
```
west build -p -b intel_adsp_ace15_mtpm -T tests/subsys/logging/log_core_additional/logging.add.async
```

**Expected behavior**
Build.

**Impact**
Does not build. Not a showstopper, but not sure what the impact is to us as users personally... a low priority for the project in any case.

**Logs and console output**
```
warning: LOG_BACKEND_UART (defined at subsys/logging/backends/Kconfig.uart:4) was assigned the value
'y' but got the value 'n'. Check these unsatisfied dependencies: UART_CONSOLE (=n). See
http://docs.zephyrproject.org/latest/kconfig.html#CONFIG_LOG_BACKEND_UART and/or look up
LOG_BACKEND_UART in the menuconfig/guiconfig interface. The Application Development Primer, Setting
Configuration Values, and Kconfig - Tips and Best Practices sections of the manual might be helpful
too.
...
tests/subsys/logging/log_core_additional/src/log_test.c:492:46: error: format string is not a string literal (potentially insecure) [-Werror,-Wformat-security]
                          LOG_LEVEL_INTERNAL_RAW_STRING, NULL, 0, test_msg_usr);
                                                                  ^~~~~~~~~~~~
include/zephyr/logging/log_msg.h:472:25: note: expanded from macro 'Z_LOG_MSG2_CREATE'
                           Z_LOG_STR(_level, __VA_ARGS__))
                                             ^~~~~~~~~~~
include/zephyr/logging/log_core.h:143:42: note: expanded from macro 'Z_LOG_STR'
                (Z_LOG_STR_WITH_PREFIX(__VA_ARGS__)), (__VA_ARGS__))
                                                       ^~~~~~~~~~~
include/zephyr/sys/util_macro.h:177:35: note: expanded from macro 'COND_CODE_1'
        Z_COND_CODE_1(_flag, _if_1_code, _else_code)
                                         ^~~~~~~~~~
note: (skipping 34 expansions in backtrace; use -fmacro-backtrace-limit=0 to see all)
include/zephyr/logging/log_msg.h:256:5: note: expanded from macro 'Z_LOG_MSG2_SIMPLE_CREATE'
                                __VA_ARGS__); \
                                ^~~~~~~~~~~
include/zephyr/sys/cbprintf.h:383:28: note: expanded from macro 'CBPRINTF_STATIC_PACKAGE'
                                  align_offset, flags, __VA_ARGS__)
                                                       ^~~~~~~~~~~
include/zephyr/sys/cbprintf_internal.h:511:53: note: expanded from macro 'Z_CBPRINTF_STATIC_PACKAGE'
                outlen = cbprintf_package(packaged, inlen, flags, __VA_ARGS__); \
                                                                  ^~~~~~~~~~~
tests/subsys/logging/log_core_additional/src/log_test.c:492:46: note: treat the string as an argument to avoid this
                          LOG_LEVEL_INTERNAL_RAW_STRING, NULL, 0, test_msg_usr);
                                                                  ^
                                                                  "%s", 
include/zephyr/logging/log_msg.h:472:25: note: expanded from macro 'Z_LOG_MSG2_CREATE'
                           Z_LOG_STR(_level, __VA_ARGS__))
                                             ^
include/zephyr/logging/log_core.h:143:42: note: expanded from macro 'Z_LOG_STR'
                (Z_LOG_STR_WITH_PREFIX(__VA_ARGS__)), (__VA_ARGS__))
                                                       ^
include/zephyr/sys/util_macro.h:177:35: note: expanded from macro 'COND_CODE_1'
        Z_COND_CODE_1(_flag, _if_1_code, _else_code)
                                         ^
note: (skipping 34 expansions in backtrace; use -fmacro-backtrace-limit=0 to see all)
include/zephyr/logging/log_msg.h:256:5: note: expanded from macro 'Z_LOG_MSG2_SIMPLE_CREATE'
                                __VA_ARGS__); \
                                ^
include/zephyr/sys/cbprintf.h:383:28: note: expanded from macro 'CBPRINTF_STATIC_PACKAGE'
                                  align_offset, flags, __VA_ARGS__)
                                                       ^
include/zephyr/sys/cbprintf_internal.h:511:53: note: expanded from macro 'Z_CBPRINTF_STATIC_PACKAGE'
                outlen = cbprintf_package(packaged, inlen, flags, __VA_ARGS__); \
                                                                  ^
```

**Environment (please complete the following information):**
 - OS: Linux
 - XCC-Clang
 - main as of Tue Feb 7 2023

--- a/tests/subsys/logging/log_core_additional/src/log_test.c
+++ b/tests/subsys/logging/log_core_additional/src/log_test.c
@@ -20,6 +20,8 @@
 #include <zephyr/logging/log.h>
 #include <zephyr/logging/log_output.h>
 
+#define TEST_MESSAGE "test msg"
+
 #define LOG_MODULE_NAME log_test
 LOG_MODULE_REGISTER(LOG_MODULE_NAME, LOG_LEVEL_INF);
 static K_SEM_DEFINE(log_sem, 0, 1);
@@ -29,7 +31,6 @@ ZTEST_BMEM uint32_t source_id;
 /* used when log_msg create in user space */
 ZTEST_BMEM uint8_t domain, level;
 ZTEST_DMEM uint32_t msg_data = 0x1234;
-ZTEST_DMEM char *test_msg_usr = "test msg";
 
 static uint8_t buf;
 static int char_out(uint8_t *data, size_t length, void *ctx)
@@ -489,7 +490,7 @@ ZTEST(test_log_core_additional, test_log_msg_create)
 
 		Z_LOG_MSG2_CREATE(!IS_ENABLED(CONFIG_USERSPACE), mode,
 			  Z_LOG_LOCAL_DOMAIN_ID, NULL,
-			  LOG_LEVEL_INTERNAL_RAW_STRING, NULL, 0, test_msg_usr);
+			  LOG_LEVEL_INTERNAL_RAW_STRING, NULL, 0, TEST_MESSAGE);
 
 		while (log_test_process()) {
 		}
@@ -507,15 +508,15 @@ ZTEST_USER(test_log_core_additional, test_log_msg_create_user)
 
 	z_log_msg_runtime_create(domain, NULL,
 				  level, &msg_data, 0,
-				  sizeof(msg_data), test_msg_usr);
+				  sizeof(msg_data), TEST_MESSAGE);
 	/* try z_log_msg_static_create() */
 	Z_LOG_MSG2_STACK_CREATE(0, domain, NULL,
 				level, &msg_data,
-				sizeof(msg_data), test_msg_usr);
+				sizeof(msg_data), TEST_MESSAGE);
 
 	Z_LOG_MSG2_CREATE(!IS_ENABLED(CONFIG_USERSPACE), mode,
 			  Z_LOG_LOCAL_DOMAIN_ID, NULL,
-		  LOG_LEVEL_INTERNAL_RAW_STRING, NULL, 0, test_msg_usr);
+		  LOG_LEVEL_INTERNAL_RAW_STRING, NULL, 0, TEST_MESSAGE);
 
 	while (log_test_process()) {
 	}

